test/testunusedvar.cpp

```diff
@@ -2003,6 +2007,54 @@ class TestUnusedVar : public TestFixture {
         ASSERT_EQUALS("", errout_str());
     }
 
+    void structMemberDefaultEq1() { // #12177
+        checkStructMemberUsageP("struct S\n"
+                                "{\n"
+                                "    bool operator==(const S&) const = default;\n"
+                                "    int i{0};\n"
+                                "};\n"
+                                "void f()\n"
+                                "{\n"
+                                "    S s;\n"
+                                "    S s1;\n"
+                                "    if (s == s1) {}\n"
+                                "}\n");
+        ASSERT_EQUALS("", errout_str());
+    }
+
+    void structMemberDefaultEq2() {
+        checkStructMemberUsageP("struct S\n"
```

Comment:
Imho, the number of lines in the test code could be reduced (with reasonable formatting).

```
struct S {
    int {0};
};
void f(S s1, S s2) {
    if (s1 == s2) {}
}
```